Dockerfile

```diff
@@ -1,60 +1,43 @@
-FROM ruby:2.6.3 as base
+FROM psul/ruby:2.6.3-node-12 as base
 
-WORKDIR /app
-
-ENV TZ=America/New_York
-
-## NodeJS
-ENV NODE_VERSION 12.9.1
-RUN mkdir /usr/local/nvm
-ENV NVM_DIR /usr/local/nvm
-SHELL ["/bin/bash", "-o", "pipefail", "-c"]
-RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash
-ENV NODE_PATH $NVM_DIR/v$NODE_VERSION/lib/node_modules
-ENV PATH $NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH
-
-
-RUN . $NVM_DIR/nvm.sh \
-    && nvm install $NODE_VERSION \
-    && nvm alias default $NODE_VERSION \
-    && nvm use default
-
-RUN npm install -g yarn@1.19.1
-
-### Envconsul
-RUN curl -Lo /tmp/envconsul.zip https://releases.hashicorp.com/envconsul/0.9.0/envconsul_0.9.0_linux_amd64.zip && \
-    unzip /tmp/envconsul.zip -d /bin && \
-    rm /tmp/envconsul.zip
-
-RUN useradd -u 10000 app -d /app
+RUN useradd -u 2000 app -d /app
+RUN mkdir /app/tmp
 RUN chown -R app /app
 USER app
 
 COPY Gemfile Gemfile.lock /app/
+COPY --chown=app vendor/ vendor/
 RUN gem install bundler:2.0.2
-RUN bundle install --path vendor/bundle
+RUN bundle install --path vendor/bundle && \
+  rm -rf /app/.bundle/cache && \
+  rm -rf /app/vendor/bundle/ruby/*/cache
+
 
 COPY package.json yarn.lock /app/
-RUN yarn --frozen-lockfile
+RUN yarn --frozen-lockfile && \
+  rm -rf /app/.cache && \
+  rm -rf /app/tmp
+
 
 COPY --chown=app . /app
 
 CMD ["./entrypoint.sh"]
 
-# Build targets for testing
-FROM base as rspec
-CMD [ "/app/bin/ci-rspec" ]
+# Final Target
+FROM base as production
 
-FROM base as eslint
-CMD [ "/app/bin/ci-eslint" ]
+RUN RAILS_ENV=production \
```

Comment:
@whereismyjetpack this is only for the build that deploys, not the one that's used for running rspec, right? RAILS_ENV would need to be "test" for the build that runs rspec.

Reply:
for tests we only run the `base` target. the last target is to bake the assets, and ship the image!